webapp-user/src/App.js

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (C) 2017-2018 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+import React, { Component } from 'react';
+import Header from './components/Header'
+import SystemUserForm from './components/SystemUserForm'
+import If from './components/If'
+import AlertBox from './components/AlertBox'
+import Models from './models/models'
+import Assertion from './models/assertion'
+import {sectionFromHash, handleError} from './models/Utils'
+import createHistory from 'history/createBrowserHistory'
+import './sass/App.css';
+
+const history = createHistory()
+
+class App extends Component {
```

Comment:
Yes, the "createClass" approach is standard Javascript style (ES5?) and I'm moving towards the ES6 style as that's the future.

This looks like a good intro: https://daveceddia.com/react-es5-createclass-vs-es6-classes/

Here's the react docs on components: https://facebook.github.io/react/docs/components-and-props.html

Just to confuse you even more, they recommend using functions as components wherever possible i.e. whenever they don't need state: https://hackernoon.com/react-stateless-functional-components-nine-wins-you-might-have-overlooked-997b0d933dbc

I'm still trying to switch my brain to use functions as components, which is why you aren't seeing too many of them!